pkg/endpoint/api.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016-2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"bytes"
+
+	"github.com/cilium/cilium/api/v1/models"
+)
+
+func ValidPatchTransitionState(state models.EndpointState) bool {
+	switch string(state) {
+	case "", StateWaitingForIdentity, StateReady:
+		return true
+	}
+	return false
+}
+
+// ProcessChangeRequest handles the update logic for performing a PATCH operation
+// on a given Endpoint. Returns the reason which will be used for informational
+// purposes should a caller choose to try to regenerate this endpoint, as well
+// as an error if the Endpoint is being deleted, since there is no point in
+// changing an Endpoint if it is going to be deleted.
+func (e *Endpoint) ProcessChangeRequest(epTemplate *models.EndpointChangeRequest, newEp *Endpoint) (string, error) {
```

Comment:
Should `endpoint.NewEndpointFromChangeModel` also move here since it's strictly an API related thing?

Reply:
Eventually, I'd like to do that. However, I'd have to then perform an endpointmanager lookup in the endpoint to see if the endpoint exists, which I can't do right now because the endpointmanager cannot be provided as a parameter anywhere.